### ISSUE TITLE
Implementation issue with model ResNet50-places365

### ISSUE THREAD

**Issue by GrumpyZhou** [CSAILVision/places365#32]
Hi,

I downloaded the two ResNet model from you:
1.  [ResNet152-places365 trained from scratch using Torch](http://places2.csail.mit.edu/models_places365/resnet152_places365.t7)
2. [ResNet50-places365 trained from scratch using Torch](http://places2.csail.mit.edu/models_places365/resnet50_places365.t7)

As I use tensorflow, so I have to parse the model first...(I use pytorch torch.utils.serialization.load_lua()).   
And I was confused by the structure of **ResNet50**, because I saw it does batch normalization after element-wise addition of original and shortcut.  

As an example, I pasted the representation of the 2nd block of conv2_x, which corresponds to part(in your code): 
````
layer {
	bottom: "res2a_branch1"
	bottom: "res2a_branch2c"
	top: "res2a"
	name: "res2a"
	type: "Eltwise"
}
````
````
nn.Sequential {
  [input -> (0) -> (1) -> output]
  (0): torch.legacy.nn.ConcatTable.ConcatTable {
    input
      |`-> (0): nn.Sequential {
      |      [input -> (0) -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> (8) -> output]
      |      (0): nn.SpatialBatchNormalization  
      |      (1): TorchObject(cudnn.ReLU, {'output': [torch.FloatTensor with no dimension]
      |      , '_type': 'torch.FloatTensor', 'train': True, 'inplace': True, 'gradInput': [torch.FloatTensor with no dimension]
      |      })
      |      (2): TorchObject(cudnn.SpatialConvolution, {'groups': 1, '_type': 'torch.FloatTensor', 'weight': 

... (3) - (5)
      |      (6): nn.SpatialBatchNormalization
      |      (7): TorchObject(cudnn.ReLU, {'output': [torch.FloatTensor with no dimension]
      |      , '_type': 'torch.FloatTensor', 'train': True, 'inplace': True, 'gradInput': [torch.FloatTensor with no dimension]
      |      })
      |      (8): TorchObject(cudnn.SpatialConvolution, {'groups': 1, '_type': 'torch.FloatTensor', 
... 
      |      [torch.FloatTensor of size 256x64x1x1]
      |      , 'train': True, 'kW': 1, 'padW': 0, 'output': [torch.FloatTensor with no dimension]
      |      , 'dW': 1, 'nInputPlane': 64})
      |    }
      |`-> (1): nn.Identity
       +. -> output
  }
  (1): nn.CAddTable
}
````

But according to your code,  `(0): nn.SpatialBatchNormalization` should be before addition, right? After checking model  **ResNet152**,
I found its implementation seems more consistent with the code in [deploy_resnet152_places365.prototxt](https://github.com/CSAILVision/places365/blob/master/deploy_resnet152_places365.prototxt), as following:
````
nn.Sequential {
  [input -> (0) -> (1) -> (2) -> output]
  (0): torch.legacy.nn.ConcatTable.ConcatTable {
    input
      |`-> (0): nn.Sequential {
      |      [input -> (0) -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> output]
      |      (0): TorchObject(cudnn.SpatialConvolution, {'groups': 1, '_type': 'torch.FloatTensor', 'weight': 

... (1) - (6)

      |      , 'dW': 1, 'nInputPlane': 64})
      |      (7): nn.SpatialBatchNormalization
      |    }
      |`-> (1): nn.Identity
       +. -> output
  }
  (1): nn.CAddTable
  (2): TorchObject(cudnn.ReLU, {'output': [torch.FloatTensor with no dimension]
````

The confirmation is important for me to reuse your model, thanks!

**Comment by gunshi** [comment]
Hi @GrumpyZhou , were you able to successfully use the places-cnns in tensorflow?
